drivers/adc/adc_smartbond_gpadc.c

```diff
@@ -251,26 +299,94 @@ static int adc_smartbond_read_async(const struct device *dev,
 }
 #endif /* CONFIG_ADC_ASYNC */
 
-static int adc_smartbond_init(const struct device *dev)
+static int gpadc_smartbond_resume(const struct device *dev)
 {
-	int err;
-	struct adc_smartbond_data *data = dev->data;
+	int ret;
 	const struct adc_smartbond_cfg *config = dev->config;
 
+	da1469x_pd_acquire(MCU_PD_DOMAIN_PER);
+
 	GPADC->GP_ADC_CTRL_REG = GPADC_GP_ADC_CTRL_REG_GP_ADC_EN_Msk;
```

Comment:
Subroutine should make sure that GPADC LDO is settled before a conversion is initiated. I guess, the risk is increased when in runtime mode as device resuming and channel conversion should be requested back to back. A delay of 22uS should cover both XTAL32M and PLL96 system clocks as described in DS:
 
<img width="549" alt="GPADC_LDO_SETTLING" src="https://github.com/zephyrproject-rtos/zephyr/assets/13284615/b1e0d0b4-197b-444d-8422-753c27bcc978">

Problem is that a delay cannot be less that OS tick's granularity so, there should be a penalty of unnecessary latency. In fact, there is a way to trick the kernel as described in [k_usleep ](https://docs.zephyrproject.org/3.0.0/reference/kernel/threads/index.html#group__thread__apis_1ga48d4b041790454da4d68ac8711f29657)but this should raise other issues. Other solution would be to introduce a custom delay subroutine that should execute the appropriate number of `nop` instructions.

Reply:
This delay is built in to the ADC, so it will wait before starting to convert. I'll look into adding a routine that reads out the current clock speed and sets this value accordingly

Reply:
Implemented as per the datasheet